### PR TITLE
more nfrs, hazard consideration, formatting, missing sections

### LINKED ISSUE #38: Finishing Steps of SRS
**Details**
The remaining work to complete for the SRS

**Desired Changes**

- [x] moving NFRs into the appropriate sections (3.4,3.5,3.6)
- [x] throw requirements into proof of concept
- [x] add everything into latex
- [x] document overview
- [x] user characteristics
- [x] 2.1 product perspective
- [x] 2.3 product constraints
- [x] 2.4 user characteristics
- [x] 2.6 apportioning of requirements
- [x] add remaining diagrams into latex
- [x] merge everything
- [x] remove wss

### LINKED ISSUE #50: Peer Review - SRS missing definitions
**Artifact Under Review**
*Section 1.3* - Definitions, Acronyms, and Abbreviations

**Team Number for Team Doing the Review**
Team 18

**Description of Issue**
The "Definitions, Acronyms, and Abbreviations" section lists several key terms, but many essential AR-related terms such as "Realm," "Tour," and "Sub-Realm" are not defined. This creates ambiguity for readers unfamiliar with these concepts, making it harder to follow subsequent sections.
*Suggestion:* Expand the section to include definitions for AR-specific terms and key app-related concepts. This will ensure clarity and consistency throughout the document.
*Rubric Criteria:* Document Content (1/3).

### LINKED ISSUE #54: Peer Review - [FR] Behavioral Descriptions
**Artifact Under Review**
Section 3.2 - Functional Requirements (Behavioral Overview)

**Team Number for Team Doing the Review**
Team 18

**Description of Issue**
The document does not provide sufficient detail on how undesired events will be handled, such as app crashes or failed AR object placement. There are no behavioral descriptions for handling these events, which could impact the user experience during system failures.

_Suggestion_: Add behavioral descriptions for how the system should respond to undesired events like crashes, network errors, or failed object placements to ensure robust error handling.

_Rubric Criteria_: Document Content (2/3), What not How (1/3).

### PR DESCRIPTION
**Pull request by 1Midnight1**
**a lot of line changes are just from auto-formatting aligning nested sections with tabs, no semantic changes were made to use cases**

### New
- Explanation under certain headings
- missing definitions addedd
- added **Usability** section for NFRs
- added **Safety** section for some NFRs in hazard analysis
- added **Databases** FR section for DB reqs that were functional
- Hazard Analysis requirements added (with rationale and fit criteria for nfrs)
    - SAR-1 -> Safety
    - SAR-2 -> Safety (modified, as upon thinking about it, our app should never really be presenting bright flashes or loud noises)
    - SAR-3 -> Left out (following from the above modification)
    - SAR-4 -> Realm Interface FR
    - SAR-5 -> Realm Interface FR
    - SAR-6 -> Maps FR
    - SER-1 -> Security
    - SER-2 -> Already Exists (PS-FR1, PS-FR2)
    - SER-3 -> Databases FR
    - SER-4 -> Security (Generalized)
    - SER-5 -> Profile FR
    - SER-6 -> Questionably not a req of the system
    - ROR-1 -> Object Placement FR
    - ROR-2 -> Maintainability
    - ROR-3 -> Databases FR
    - ROR-4 -> Reliability 
    - ROR-5 -> Already Exists (MP-FR8)
    - ROR-6 -> Performance
    - ROR-7 -> Realm Interface FR - needs expanding
    - PVR-1 - Maps FR
- Filled in Cost, Deadline, and Proof of Concept Sections
- New Object Placement FRs
- Added atleast one NFR to sections with none

### Changed
- auto-formatted to fix some formatting issues
- all references to sub-realms are now refs/hyperrefs to the definition or requirements section as appropriate
- makefile updated to use bibtex (fixed references)
- interface names and descriptions moved to ui section
- made some definitions more specific
- *groups* changed to *sub-realms* where relevant
- fixed formatting in some OUI, OS, and POG requirements sections
- NFRs moved to relevant sections with rationale and fit criteria added (with the exception of some performance reqs)
    - MP-NFR1 -> Performance
    - MP-NFR2 -> Changed to FR
    - MP-NFR3 -> Generalized to a Settings FR
    - IV-NFR1 -> Performance
    - IV-NFR2 -> Changed to FR
    - IV-NFR3 -> Changed to FR
    - OS-NFR1 -> Performance
    - OUI-NFR1 -> Performance
    - POG-NFR1 -> Performance
    - POG-NFR2 -> Changed to FR
    - POG-NFR3 -> Usability (Generalized)
    - POG-NFR4 -> Usability (Generalized)

### Removed
- duplicate assumptions

### Addresses
Closes #50
Closes #54 
Closes #38 
Progresses towards #43
Progresses towards #47

### CODE REVIEW

**Comment by Avanish-a** [comment]
The commit names should include the relative file path, so in this case, you can add `docs/SRS-IEEE`.

**Comment by 1Midnight1** [comment]
> The commit names should include the relative file path, so in this case, you can add `docs/SRS-IEEE`.

I don't think we set this standard anywhere, and we certainly haven't been following this thus far

**Comment by Avanish-a** [comment]
> I don't think we set this standard anywhere, and we certainly haven't been following this thus far

We can start following it now. It helps with the traceability of commits to files.

**Comment by 1Midnight1** [comment]
@Avanish-a please provide specifics on the problems introduced by these changes

File: docs/SRS-IEEE/Makefile [hunk 1]
```diff
@@ -8,8 +8,8 @@ BUILDTEX=$(TEX) $(PROJECT).tex
 
 all:
 	$(BUILDTEX)
-	# $(BIBTEX) $(PROJECT)
-	# $(BUILDTEX)
+	$(BIBTEX) $(PROJECT)
+	$(BUILDTEX)
```
[the remark this answers is not in the document]
Comment (1Midnight1) [reply docs/SRS-IEEE/Makefile:12]: Fixes references when building with makefile

File: docs/SRS-IEEE/SRS.tex [hunk 2]
```diff
@@ -547,20 +545,21 @@ \subsubsection{Profile Screen}
...
-    \item The user must be able to delete a group, and all associated data should be removed.
-    \item All users within a group must be able to interact with group-specific AR content, including shared AR objects and experiences.
+    \item The user must be able to create a new sub-realm with a sub-realm name, description, and also allow inviting members.
+    \item Allow users to add or remove members from a sub-realm.
+    \item The user must be able to edit sub-realm settings, including sub-realm name and description.
+    \item The user must be able to delete a sub-realm, and all associated data should be removed.
+    \item All users within a sub-realm must be able to interact with sub-realm-specific AR content, including shared AR objects and experiences.
 \end{enumerate}
 
 \subsubsection{Friends screen}
```
[the remark this answers is not in the document]
Comment (1Midnight1) [reply docs/SRS-IEEE/SRS.tex:565]: Not a part of pr changes

File: docs/SRS-IEEE/SRS.tex [hunk 3]
```diff
@@ -71,28 +72,20 @@ \subsection{Definitions, Acronyms and Abbreviations}
...
+    \item \labeltext{AR object cluster}{def:ar_obj_cls}: A group of AR objects present within a confined area.
     \item \labeltext{Users}{def:user}: A term for anyone who uses the app.
-    \item \labeltext{Organization users}{def:org_user}: Users who belong to an organization that has the ability to create tours. They are affiliated with a particular approved organization who have the ability to modify tours within their organization’s domain.
+    \item \labeltext{Organization users}{def:org_user}: Users who belong to an organization that have the ability to create tours. They are affiliated with a particular approved organization who have the ability to modify tours within their organization’s domain.
     \item \labeltext{General users}{def:gen_user}: Users that can access most of the app functionalities except creating tours.
     \item \labeltext{Admins}{def:admin}: People who have access to nothing else but the admin interface within the app.
-    \item \labeltext{AR object clusters}{def:ar_obj_cls}: A group of AR objects present within a confined area.
-\end{itemize}
-
-\subsubsection{Screen Names}
```
Comment (rafeymalik1) [inline docs/SRS-IEEE/SRS.tex:82]: Why did you remove the screen names' definitions?

Comment (1Midnight1) [reply docs/SRS-IEEE/SRS.tex:82]: it was moved to the user interfaces section and renamed interface names or something

Comment (rafeymalik1) [reply docs/SRS-IEEE/SRS.tex:82]: Resolved

**Review by Avanish-a — APPROVED** [review]

**Review by rafeymalik1 — APPROVED** [review]
Lots of formatting errors and other issues have been resolved, along with some new content added. Great work!

**Pull request merged**